src/D2L.Bmx/Okta/OktaClient.cs

```diff
@@ -87,16 +98,12 @@ await resp.Content.ReadAsStreamAsync(),
 			);
 		}
 
-		string org = _organization ?? "unknown";
-		throw new BmxException( $"""
-			Okta authentication for user '{username}' in org '{org}' failed.
-			Check if org, user, and password is correct.
-			""" );
+		return new AuthenticateResponse.Failure( resp.StatusCode );
```

Comment:
This is new.

The `_organization` field was saved in the original `OktaApi` class for the sole purpose of constructing this error message. This isn't great, however.
The `OktaApi` / `Okta*Client` classes are meant to be low level and only care about talking to Okta. Constructing informative error messages with rich context info is more of a "business concern", and better left to higher level code, especially when not doing so complicates the low level classes (e.g. saving an otherwise useless extra field).